web/src/components/tasks/search-business-name/SearchBusinessNameForm.test.tsx

```diff
@@ -9,102 +14,126 @@ import {
   searchAndReject,
   searchButton,
 } from "@/test/helpers/helpersSearchBusinessName";
-import { useMockRoadmap } from "@/test/mock/mockUseRoadmap";
-import { useMockProfileData, useMockUserData } from "@/test/mock/mockUseUserData";
-import { fireEvent, render, screen, within } from "@testing-library/react";
-import { ReactElement } from "react";
+import {
+  castPublicFilingLegalTypeToFormationType,
+  generateFormationFormData,
+  generateUserData,
+  ProfileData,
+  PublicFilingLegalType,
+} from "@businessnjgovnavigator/shared";
+import * as materialUi from "@mui/material";
+import { fireEvent, screen, within } from "@testing-library/react";
+
+function mockMaterialUI(): typeof materialUi {
+  return {
+    ...jest.requireActual("@mui/material"),
+    useMediaQuery: jest.fn(),
+  };
+}
 
+jest.mock("@mui/material", () => mockMaterialUI());
 jest.mock("@/lib/data-hooks/useUserData", () => ({ useUserData: jest.fn() }));
 jest.mock("@/lib/data-hooks/useRoadmap", () => ({ useRoadmap: jest.fn() }));
+jest.mock("@/lib/data-hooks/useDocuments");
+jest.mock("next/router", () => ({ useRouter: jest.fn() }));
 jest.mock("@/lib/api-client/apiClient", () => ({
+  postBusinessFormation: jest.fn(),
+  getCompletedFiling: jest.fn(),
   searchBusinessName: jest.fn(),
 }));
+
 const mockApi = api as jest.Mocked<typeof api>;
 
-const Config = getMergedConfig();
-const onChange = jest.fn();
+describe("SearchBusinessNameForm", () => {
+  const displayContent = {
+    formationDbaContent: generateFormationDbaContent({}),
+  };
 
-describe("<SearchBusinessNameForm />", () => {
   beforeEach(() => {
     jest.resetAllMocks();
-    useMockUserData({});
-    useMockRoadmap({});
+    useSetupInitialMocks();
   });
 
-  const availableText = "AVAILABLE!";
-  const unavailableText = "UNAVAILABLE!";
-  const FakeAvailable = (): ReactElement => {
-    return <div>{availableText}</div>;
-  };
-  const FakeUnavailable = (): ReactElement => {
-    return <div>{unavailableText}</div>;
+  const getPageHelper = async (initialProfileData?: Partial<ProfileData>): Promise<FormationPageHelpers> => {
+    const profileData = generateFormationProfileData({
+      ...initialProfileData,
+      legalStructureId: "limited-liability-company",
+      businessPersona: "FOREIGN",
+    });
+    const formationData = {
+      formationFormData: generateFormationFormData(
+        {},
+        {
+          legalStructureId: castPublicFilingLegalTypeToFormationType(
+            profileData.legalStructureId as PublicFilingLegalType,
+            profileData.businessPersona
+          ),
+        }
+      ),
+      formationResponse: undefined,
+      getFilingResponse: undefined,
+      completedFilingPayment: false,
+      businessNameAvailability: undefined,
+      dbaBusinessNameAvailability: undefined,
+      lastVisitedPageIndex: 0,
+    };
+    const page = preparePage(generateUserData({ profileData, formationData }), displayContent);
+    return page;
```

Comment:
It seems like this function is used in like all of these tests. Could it be extracted as a helper?

Reply:
I started to pull this out and use same `preparePage` that's being exported by `testHelpers.tsx`. It's being used extensively which I didn't realize. I need to modify the function slightly to account for formationData and not just formationFormData. It touches enough that I think it's worth including separate from this PR.